packages/documentation-framework/layouts/sideNavLayout/sideNavLayout.js

```diff
@@ -87,6 +91,15 @@ const HeaderTools = ({
                 document.querySelector('html').classList.toggle('pf-v5-theme-dark')} />
             </ToolbarItem>
           )}
+          {hasRTLSwitcher && (
+            <ToolbarItem>
+              <Switch id="ws-rtl-switch" label={isRtl ? 'RTL' : 'LTR'} defaultChecked={false} onChange={() =>{
+                const examplePreviews = document.querySelectorAll('.ws-example .ws-preview');
+                examplePreviews.forEach(ex => ex.classList.toggle('pf-v5-m-dir-rtl'));
```

Comment:
Since this fires from the switch per page, it's easy for the switch's checked state to get out of sync with the examples. For example - go to the home page or some component page and switch RTL on, then go to another component page and the switch is checked but the examples are LTR. What if the switch just toggled some bool (`rtlEnabled` or something) the example example block had a conditional in it spit out the rtl class if `rtlEnabled`?